bin/handler-slack-multichannel.rb

```diff
@@ -335,6 +356,7 @@ def translate_status
       2 => :CRITICAL,
       3 => :UNKNOWN
     }
-    status[check_status.to_i]
+    # When the sensu check status is out of bounds, don't hide the status but show it for detailed info
+    status.fetch(check_status.to_i, "OUT OF BOUNDS: #{check_status.to_i}")
```

Comment:
I don't know that this is the right approach, I think we should just fall back to an `unknown`. See previous comment about merging the functionality via a library method.

Reply:
I do not entirely agree with having `OUT OF BOUNDS` messages declared as `UNKNOWN`. An UNK in nagios-style means the check code has pruposely set this exit status and reflects the purpose of the programmer to explicitly tell the operator there is no way to know the state of the service checked.
An OOB from nagios standards means that the check script has exited somewhere in a nasty way. Typical cases are uncaught bad curl exits, missing files, missing permissions (on check binary or files to read) and a myriad of them. The very reason to have OOB is to let the operator know that the check itself is failing, rather than working in an expected manner and not able to monitor a service or resource.
I can see how it would make sense to avoid duplicity here, but you could argue likewise about the whole having a `def translate_status` and a `def color` which are based on the same input. We could return both in a hash and access the key we want specifically in the code.
In any case I strongly believe having OOB explicitly defined and the exit code returned to the operator for visibility is crucial to understand what is going on with the service monitored.

Reply:
I agree with your point about `translate_status` and `color` duplication, I did not initially write that bit. I added in the error handling to return a status (and corresponding color) which is understood in the sensu ecosystem in https://github.com/sensu-plugins/sensu-plugins-slack/pull/62. More often than not changes to existing plugins are made within the confines of how it already works. I did not use that slack handler at the time so I did not really have anywhere to test it. I made the smallest and safest change that solved the problem someone was facing.

For the moment I think we should stick to the status[1] of `unknown` for non `[0-2]` codes because there is lack of precedence in the ecosystem. In ~5 years of consuming, contributing, and maintaining sensu I have yet to see anyone follow this convention you are proposing. While we mirror certain aspects of nagios I don't want to hold ourselves to all of their conventions as it ties our hands. I am not opposed to it in general but I think we should put out an [RFC](https://github.com/sensu-plugins/community/issues/new) and get community feedback before declaring a new convention to be followed. I am fine with adding some additional output to tell the operator more in the meantime.

[1]: leaving the status code intact

Reply:
You are right that it probably make sense to open this discussion to the whole community. I still advocate for my proposed convention but I will move the conversation to there for openness and greater feedback.

Reply:
Ok, added [here] (https://github.com/sensu-plugins/community/issues/127)

Reply:
Regarding the de-duplication of `translate_status` and `color`, I will check if this would mean lots of further changes. If it doesn't I will go ahead with it, otherwise I will try to keep it simple for now.

Reply:
Should be resolved bu https://github.com/sensu-plugins/sensu-plugins-slack/pull/82/commits/fac988a26e7ca46362ab7ff934c8336c8ddd84c1, please review